### ISSUE TITLE
Persistent build failed in -If resumes at one task behind the failed

### ISSUE THREAD

**Issue by nightroman** [nightroman/Invoke-Build#90]
This defect was discovered on working with [Tasks/Ask](https://github.com/nightroman/Invoke-Build/tree/master/Tasks/Ask). Namely, if a persistent build uses ask tasks and a user hits <kbd>Ctrl+C</kbd> when prompted then the build stops as expected. But then it resumes not at the failed prompt but at the previous task which was already done and should not be repeated on resuming.

It turns out we save checkpoints after `-If`. This should be done before in order to solve the problem.

NB
Why we save checkpoints in the beginning of a task. This may look unusual because "normally" checkpoints are saved in the end of successful something. Suppose we use the "normal" way. Then imagine, the very first task fails. In this case, the checkpoint is not created at all and resuming is not possible. This may be confusing for a user: when temporary problems are fixed, the failed build is expected to be *resumed* regardless of which task failed before, first or not.
